From be96ffaae1e9cdbc53151ba1797bf6dc111c5dae Mon Sep 17 00:00:00 2001
From: Will Dietz <will.dietz@sifive.com>
Date: Fri, 10 May 2024 12:43:46 -0500
Subject: [PATCH] dylib-lit-paths

lit.cfg.py only, leave Unit as-is for now.
---
 test/lit.cfg.py | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/test/lit.cfg.py b/test/lit.cfg.py
index 009a561a5..abbf72a60 100644
--- a/test/lit.cfg.py
+++ b/test/lit.cfg.py
@@ -90,3 +90,31 @@ if config.slang_frontend_enabled:
   tools.append('circt-verilog')
 
 llvm_config.add_tool_substitutions(tools, tool_dirs)
+
+
+def find_shlibpath_var():
+    if platform.system() in ["Linux", "FreeBSD", "NetBSD", "OpenBSD", "SunOS"]:
+        yield "LD_LIBRARY_PATH"
+    elif platform.system() == "Darwin":
+        yield "DYLD_LIBRARY_PATH"
+    elif platform.system() == "Windows":
+        yield "PATH"
+    elif platform.system() == "AIX":
+        yield "LIBPATH"
+
+
+for shlibpath_var in find_shlibpath_var():
+    # in stand-alone builds, shlibdir is clang's build tree
+    # while llvm_libs_dir is installed LLVM (and possibly older clang)
+    shlibpath = os.path.pathsep.join(
+        (
+            config.circt_shlib_dir,
+            config.environment.get(shlibpath_var, ""),
+        )
+    )
+    config.environment[shlibpath_var] = shlibpath
+    break
+else:
+    lit_config.warning(
+        "unable to inject shared library path on '{}'".format(platform.system())
+    )
-- 
2.39.3 (Apple Git-146)

